tests/upload_files.rs

```diff
@@ -535,3 +535,46 @@ fn assert_file_contents(file_path: &Path, contents: &str) {
     let file_contents = std::fs::read_to_string(file_path).unwrap();
     assert!(file_contents == contents)
 }
+
+#[cfg(unix)]
+#[rstest]
+#[case(server(&["-u"]), 0o600)]
+#[case(server(&["-u", "--chmod", "660"]), 0o660)]
+#[case(server(&["-u", "--chmod", "644"]), 0o644)]
+fn chmod(#[case] server: TestServer, #[case] expected_mode: u32) -> Result<(), Error> {
```

Comment:
Seems like a good test but please add a doc comment explaining the intention. I'd also love to see a second test that tests the chmod error in case you don't have permissions for the chmod but I recognize it might be hard. If you don't find an easy way to test that, it's fine.